code/game/objects/items/stacks/sheets/sheet_types.dm

```diff
@@ -57,6 +57,20 @@ var/global/list/datum/stack_recipe/metal_recipes = list ( \
 	flags = CONDUCT
 	origin_tech = "materials=1"
 
+/obj/item/stack/sheet/metal/attackby(obj/item/weapon/W, mob/living/user)
+	..()
+	if(istype(W, /obj/item/weapon/paper/talisman/construction))
```

Comment:
This was also my first approach but hitting metal with (my version)) the talisman wouldn't get a reaction. I'll try it again with your code but I spent at least 30 minutes trying all kind of afterattack code and never got it to work.